lib/src/repo.rs

```diff
@@ -660,6 +670,31 @@ impl MutableRepo {
         (self.index, self.view.into_inner())
     }
 
+    pub fn resolve_change_id_prefix<'a>(
```

Comment:
Optional: It might be nice to leave a comment explaining that this is different from and much slower than the ReadonlyRepo implementation

Reply:
I think the TODO comment copied from revset.rs implies that. It's not perfect, but we can easily deduce that.